lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -1880,10 +1880,46 @@ TypeSystemSwift::TypeAllocationStrategy
 TypeSystemSwiftTypeRef::GetAllocationStrategy(opaque_compiler_type_t type) {
   return m_swift_ast_context->GetAllocationStrategy(ReconstructType(type));
 }
+
 CompilerType TypeSystemSwiftTypeRef::CreateTupleType(
     const std::vector<TupleElement> &elements) {
-  return m_swift_ast_context->CreateTupleType(elements);
+  auto impl = [&]() -> CompilerType {
+    using namespace swift::Demangle;
+    Demangler dem;
+    auto *tuple = dem.createNode(Node::Kind::Tuple);
+    for (const auto &element : elements) {
+      auto *tuple_element = dem.createNode(Node::Kind::TupleElement);
+      tuple->addChild(tuple_element, dem);
+
+      if (!element.element_name.IsEmpty()) {
+        auto *name = dem.createNode(Node::Kind::TupleElementName,
+                                    element.element_name.GetStringRef());
+        tuple_element->addChild(name, dem);
+      }
+
+      auto *type = dem.createNode(Node::Kind::Type);
+      tuple_element->addChild(type, dem);
+      auto *element_type = GetDemangledType(
+          dem, element.element_type.GetMangledTypeName().GetStringRef());
+      type->addChild(element_type, dem);
+    }
+    return RemangleAsType(dem, tuple);
+  };
+  {
+    auto result = impl();
+    if (!m_swift_ast_context)
+      return result;
+    bool equivalent =
+        Equivalent(result, m_swift_ast_context->CreateTupleType(elements));
```

Comment:
Looks good — the only thing that's missing is wrapping the comparison code in #ifndef NDEBUG